File: 392. Is Subsequence.cpp
class Solution {

public:

    bool isSubsequence(string s, string t) {

        int s_p=0, t_p=0;

        int s_len=s.length(), t_len=t.length();

        while(s_p<s_len && t_p<t_len){

            if(s[s_p]== t[t_p]){

                s_p++;

                t_p++;

            }

            else

                t_p++;

        }

        return s_p==s_len;

    }

};

File: 392. Is Subsequence.py
class Solution(object):

    def isSubsequence(self, s, t):

        """

        :type s: str

        :type t: str

        :rtype: bool

        """

        s_len, t_len= len(s), len(t)

        s_p, t_p = 0, 0

        while(s_p<s_len and t_p<t_len):

            if(s[s_p]==t[t_p]):

                s_p += 1

                t_p += 1

            else:

                t_p += 1

        return s_p == s_len

        
